everestrs/everestrs/build.rs

```diff
@@ -0,0 +1,89 @@
+use std::env;
+use std::path::{Path, PathBuf};
+
+struct Libraries {
+    everestrs_sys: PathBuf,
+    framework: PathBuf,
+}
+
+fn find_everest_workspace_root() -> PathBuf {
+    let mut cur_dir =
+        PathBuf::from(env::var("CARGO_MANIFEST_DIR").expect("always set in build.rs execution"));
+
+    // A poor heuristic: We traverse the directories upwards until we find a directory called
+    // everest-framework and hope that this is the EVerest workspace.
+    while cur_dir.parent().is_some() {
+        cur_dir = cur_dir.parent().unwrap().to_path_buf();
+        if cur_dir.join("everest-framework").is_dir() {
+            return cur_dir;
+        }
+    }
+    panic!("everstrs is not build within an EVerest workspace.");
+}
+
+/// Returns the Libraries path if this is a standalone build of everest-framework or None if it is
+/// not.
+fn find_libs_in_qwello_framework(root: &Path) -> Option<Libraries> {
+    let everestrs_sys = root.join("everest-framework/build/everestrs/libeverestrs_sys.a");
+    let framework = root.join("everest-framework/build/lib/libframework.so");
+    if everestrs_sys.exists() && framework.exists() {
+        Some(Libraries {
+            everestrs_sys,
+            framework,
+        })
+    } else {
+        None
+    }
+}
+
+/// Returns the Libraries path if this is an EVerest workspace where make install was run in
+/// everest-core/build or None if not.
+fn find_libs_in_qwello_core_build_dist(root: &Path) -> Option<Libraries> {
+    let everestrs_sys = root.join("everest-core/build/dist/lib/libeverestrs_sys.a");
+    let framework = root.join("everest-core/build/dist/lib/libframework.so");
+    if everestrs_sys.exists() && framework.exists() {
+        Some(Libraries {
+            everestrs_sys,
+            framework,
+        })
+    } else {
+        None
+    }
+}
+
+/// Takes a path to a library like `libframework.so` and returns the name for the linker, aka
+/// `framework`
+fn libname_from_path(p: &Path) -> String {
+    let mut name = p
+        .file_stem()
+        .and_then(|os_str| os_str.to_str())
+        .unwrap_or("")
+        .to_string();
+    name.drain(..3); // remove 'lib' prefix.
```

Comment:
https://doc.rust-lang.org/std/string/struct.String.html#method.strip_prefix should be better and easier to understand

Reply:
good call. Done.